docs/configuration/tools/router_check.rst

```diff
@@ -0,0 +1,160 @@
+
+Validate
+  The validate fields specify the expected values and test cases to check. At least one test
+  case is required.
+
+A simple tool configuration json has one test case and is writen as follows. The test
+expects a cluster name match of "instant-server".::
+
+   [
+     {
+       "test_name: "Cluster_name_test",
+       "input":
+         {
+           ":authority":"api.lyft.com",
+           ":path": "/api/locations"
+         }
+       "_validate"
+         {
+           "cluster_name": "instant-server"
+         }
+      }
+   ]
+
+.. code-block:: json
+
+  [
+    {
+      "test_name": "...",
+      "input":
+        {
+          ":authority": "...",
+          ":path": "...",
+          ":method": "...",
+          "internal" : "...",
+          "random_value" : "...",
+          "ssl" : "...",
+          "additional_headers": [
+            {
+              "field": "...",
+              "value": "..."
+            },
+            {
+               "..."
+            }
+          ]
+        }
+      "_validate": {
+        "cluster_name": "...",
+        "virtual_cluster_name": "...",
+        "virtual_host_name": "...",
+        "host_rewrite": "...",
+        "path_rewrite": "...",
+        "path_redirect": "...",
+        "header_fields" : [
+          {
+            "field": "...",
+            "value": "..."
+          },
+          {
+            "..."
+          }
+        ]
+      }
+    },
+    {
+      "..."
+    }
+  ]
+
+test_name
+  *(required, string)* The name of a test object.
+
+input
+  *(required, object)* Input values sent to the router that determine the returned route. The :authority
+  and :path input fields are required.
+
+  :authority
+    *(required, string)* The url authority. This value along with the path parameter define
+    the url to be matched. An example authority value is "api.lyft.com".
+
+  :path
+    *(required, string)* The url path. An example path value is "/foo".
+
+  :method
+    *(optional, string)* The request method. If not specified, the default method is GET. The options
+    are GET, PUT, or POST.
+
+  internal
+    *(optional, boolean)* A flag that determines whether to set x-envoy-internal to "true".
+    If not specified, or if internal is equal to false, x-envoy-internal is not set.
+
+  random_value
+    *(optional, integer)* An integer used to identify the target for weighted cluster selection.
+    The default value of random_value is 0.
+
+  ssl
+    *(optional, boolean)* A flag that determines whether to set x-forwarded-proto to https or http.
+    By setting x-forwarded-proto to a given protocol, the tool is able to simulate the behavior of
+    a client issuing a request via http or https. By default ssl is false which corresponds to
+    x-forwarded-proto set to http.
+
+
+  additional_headers
+    *(optional, array)*  Additional headers to be added as input for route determination. The ":authority",
```

Comment:
can you please expand what the content of the array are? Similar to adding a section for https://lyft.github.io/envoy/docs/configuration/http_conn_man/route_config/route.html#adding-custom-request-headers 

Please do this for the expected output for headers